### PR TITLE
feat:SPT-466 - Added action to retrieve aws account id

### PR DESCRIPTION
**Pull request by MartinMitro**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    description: "AWS account tag used for retrieving id"
+    value: ${{ steps.retrieve.outputs.tag }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set up aws credentials
+      shell: bash
+      run: |
+        aws configure set aws_access_key_id ${{ inputs.aws-access-key }}
```
Comment (rbrisuda) [inline action.yml]: maybe different profile than default, composite action I think should not reset your default profile

**Review by rbrisuda — DISMISSED** [review]

File: action.yml (hunk 1 above)
Comment (MartinMitro) [reply action.yml]: done

**Review by mliner — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -1,2 +1,63 @@
...
+        default: 'dvc'
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ohpensource/get-aws-account-id-gh-action@0.0.0.1
+        id: aws_account
+        name: Get account id
+        env:
```
Comment (mmichna) [inline README.md]: remove ENV variable and move definition to action field

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            exit -1
+        fi
+        if (( $RESOURCES_COUNT > 1 )); then
+            echo "Multiple accounts tagged with tag ${AWS_ACCOUNT_TAG}. Either verify tag key or contact PST"
+            exit -1
+        fi
+        ACCOUNT_ID=$(jq -r '.ResourceTagMappingList[0].ResourceARN' <<< $RESOURCES | sed 's/.*\///')
+
+        echo "::set-output name=id::${ACCOUNT_ID}"
+        echo "::set-output name=tag::${AWS_ACCOUNT_TAG}"
```
Comment (mmichna) [inline action.yml]: PLS initialize **AWS_ACCOUNT_TAG** with **aws-account-tag** field value  and not depends on ENV variables

**Review by mmichna — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (MartinMitro) [reply README.md]: why?

File: action.yml (hunk 3 above)
Comment (MartinMitro) [reply action.yml]: this line was removed

File: README.md (hunk 2 above)
Comment (mmichna) [reply README.md]: because it is standard format send all values to action via input parameters and not environment variables

Comment (mmichna) [reply README.md]: and now you have duplicity **AWS_ACCOUNT_TAG** and also **aws-account-tag** - and this is not necessary

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        aws configure set role_arn ${{ inputs.aws-role-arn }} --profile organizations
+        aws configure set source_profile organizations-base --profile organizations
+    - name: Retrieve aws account id
+      id: retrieve
+      shell: bash
+      run: |
+        RESOURCES=$(aws resourcegroupstaggingapi get-resources --resource-type-filters organizations:account --tag-filters Key=${{ inputs.aws-account-tag }} --profile organizations --region us-east-1)
+        RESOURCES_COUNT=$(jq '.ResourceTagMappingList | length' <<< $RESOURCES)
+        if [ $RESOURCES_COUNT == 0 ]; then
+            echo "No account found with tag ${AWS_ACCOUNT_TAG}"
```
Comment (mmichna) [inline action.yml]: instead of **AWS_ACCOUNT_TAG** is better use **inputs.aws-account-tag**

File: README.md (hunk 2 above)
Comment (MartinMitro) [reply README.md]: I think you got it wrong, I will contact you via Teams

**Review by mmichna — APPROVED** [review]

**Review by mliner — APPROVED** [review]

**Pull request merged**